argocd/resource_argocd_application.go

```diff
@@ -20,7 +21,7 @@ func resourceArgoCDApplication() *schema.Resource {
 		Delete: resourceArgoCDApplicationDelete,
```

Comment:
CRUD must be replaced by their context aware equivalents here

Reply:
> All CRUD methods should be replaced by their context aware equivalents. I put a few notes on a couple resources to specify which.

It is gonna take some time to make those changes, it's probably better to open another PR just for those. Wdyt?

Reply:
I would prefer to tackle any leftover from the v1 SDK in that PR, and use the v2 SDK updated methods everywhere, so that no deprecated functions are still being called. Then as a second step, adding new features that the v2 SDK offer in another PR - extended TypeMap validator functions come to mind - would be a possibility.